Startup.cs

```diff
@@ -22,116 +15,12 @@ public Startup(IConfiguration configuration)
-                {
-                    var user = context.HttpContext.User;
-
-                    // Avoid displaying the select account dialog
-                    context.ProtocolMessage.LoginHint = user.GetLoginHint();
-                    context.ProtocolMessage.DomainHint = user.GetDomainHint();
-
-                    // Remove the account from MSAL.NET token cache
-                    var _tokenAcquisition = context.HttpContext.RequestServices.GetRequiredService<ITokenAcquisition>();
-                    await _tokenAcquisition.RemoveAccount(context);
-                };
-
-                // Avoids having users being presented the select account dialog when they are already signed-in
-                // for instance when going through incremental consent 
-                options.Events.OnRedirectToIdentityProvider = async context =>
-                {
-                    string login = context.Properties.GetParameter<string>(OpenIdConnectParameterNames.LoginHint);
-                    if (!string.IsNullOrWhiteSpace(login))
-                    {
-                        context.ProtocolMessage.LoginHint = login;
-                        context.ProtocolMessage.DomainHint = context.Properties.GetParameter<string>(OpenIdConnectParameterNames.DomainHint);
-
-                        // delete the loginhint and domainHint from the Properties when we are done otherwise 
-                        // it will take up extra space in the cookie.
-                        context.Properties.Parameters.Remove(OpenIdConnectParameterNames.LoginHint);
-                        context.Properties.Parameters.Remove(OpenIdConnectParameterNames.DomainHint);
-                    }
-
-                    // Additional claims
-                    const string claims = "claims";
-                    if (context.Properties.Items.ContainsKey(claims))
-                    {
-                        context.ProtocolMessage.SetParameter(claims, context.Properties.Items[claims]);
-                    }
-                };
-
-                // If you want to debug, or just understand the OpenIdConnect events, just
-                // uncomment the following line of code
-                // OpenIdConnectMiddlewareDiagnostics.Subscribe(options.Events);
-
-            });
-
-            services.AddMvc(options =>
-                {
-                    var policy = new AuthorizationPolicyBuilder()
-                        .RequireAuthenticatedUser()
-                        .Build();
-                    options.Filters.Add(new AuthorizeFilter(policy));
-                })
-                .SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+            //default system actions and options integration with dependency system
+            services.InitializeDefault(Configuration); 
+            //init AzureAd specific configuration
+            services.InitializeAuthentication(Configuration); 
```

Comment:
I like the apparent simplification
The issue here is you don't specify the token cache implementation.

Reply:
True. I can do that as well. I left the implementation of the token cache in the `Bootstrapper.cs` and focus on refactoring bits with re-issuing token and modify it via filter.

I didn't focus on the `TokenAcquisition.cs` class. That can be easily refactored and merge to Boostraper.cs for easier usage and same coding style.

Do it do that as well?

Reply:
as you prefer @bojanv 
I'm not sure we'd want to merge the TokenAcquisition.cs class (reusable Aspnet service) with Bootstrapper?
how would you surface the choice of the cache implementation? something like .WithSQLTokenCache().

I can also do it when I've merged.

Reply:
Let me refactor it and fix other suggestions / fixes as well. Will do it later this evening.

Reply:
@bojanv : did I miss the refactoring you were mentioning ^^